app/controllers/support_interface/email_log_controller.rb

```diff
@@ -2,6 +2,10 @@ module SupportInterface
   class EmailLogController < SupportInterfaceController
     def index
       @emails = Email.order(id: :desc).includes(:application_form).limit(1000)
+
+      %w[to subject mailer mail_template notify_reference application_form_id delivery_status].each do |column|
+        @emails = @emails.where(column => params[column]) if params[column]
+      end
```

Comment:
I was confused here before I saw `if params[column]` as I thought this is doing ~7 `where` calls, what do you think of bringing it in the front like:

```suggestion
      %w[to subject mailer mail_template notify_reference application_form_id delivery_status]
        .select { |column| params[column] }
        .each { |column| @emails = @emails.where(column => params[column]) }
```

Reply:
Fixed in a different way.